### PR TITLE
Add eval details template for Trusty rule

### PR DESCRIPTION
**Pull request by eleftherias**
# Summary

Add an evaluation details template for the Trusty ruletype.
This will be displayed on the UI to give the user a better understanding of why the rule failed.

Fixes https://github.com/stacklok/minder-stories/issues/38

## Change Type

***Mark the type of change your PR introduces:***

- [ ] Bug fix (resolves an issue without affecting existing features)
- [x] Feature (adds new functionality without breaking changes)
- [ ] Breaking change (may impact existing functionalities or require documentation updates)
- [ ] Documentation (updates or additions to documentation)
- [ ] Refactoring or test improvements (no bug fixes or new functionality)

# Testing

***Outline how the changes were tested, including steps to reproduce and any relevant configurations. 
Attach screenshots if helpful.***

# Review Checklist:

- [x] Reviewed my own code for quality and clarity.
- [x] Added comments to complex or tricky code sections.
- [ ] Updated any affected documentation.
- [x] Included tests that validate the fix or feature.
- [ ] Checked that related changes are merged.

### CODE REVIEW

File: internal/engine/eval/templates/templates.go [hunk 1]
```diff
@@ -25,3 +25,9 @@ import (
 //
 //go:embed vulncheckTemplate.tmpl
 var VulncheckTemplate string
+
+// TrustyTemplate is the template for evaluation details of the
+// `trusty` evaluation engine.
+//
+//go:embed trustyTemplate.tmpl
+var TrustyTemplate string
```
Comment (blkt) [inline internal/engine/eval/templates/templates.go:33]: I would document the accepted parameters as well.

```suggestion
// TrustyTemplate is the template for evaluation details of the
// `trusty` evaluation engine.
//
// This template accepts two parameters, `lowScoringPackages`
// and `maliciousPackages`, which must be list of strings.
//
//go:embed trustyTemplate.tmpl
var TrustyTemplate string
```

File: internal/engine/eval/templates/trustyTemplate.tmpl [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+Malicious packages:
+{{- range .maliciousPackages }}
+* {{ . }}
+{{- end }}
+{{- end }}
+
+{{- /* Add a newline only if both maliciousPackages and lowScoringPackages exist */ -}}
+{{- if and .maliciousPackages .lowScoringPackages}}
+{{ end -}}
+
```
Comment (blkt) [inline internal/engine/eval/templates/trustyTemplate.tmpl:11]: I believe lines 7 and 11 already add newlines, what's the difference if we remove this `if`?

**Review by blkt — DISMISSED** [review]
LGTM, I just left a couple comments.

File: internal/engine/eval/templates/trustyTemplate.tmpl (hunk 2 above)
Comment (eleftherias) [reply internal/engine/eval/templates/trustyTemplate.tmpl:11]: If we remove this `if` and there are only low scoring packages, and no malicious packages, then the message has a leading `\n`: `"\nPackages with a low Trusty score:\n* package"`. It's not a big problem, but it was bugging me.

**Review by blkt — APPROVED** [review]

**Pull request merged**